### PR TITLE
29 set up devcontainers

### LINKED ISSUE #29: Set up devcontainers
The VS Code workspace should include a devcontainers setup.

### PR DESCRIPTION
**Pull request by ehrhorn**
Closes #29 .

Added a `docker-compose` dev environment.

NOTE: Actions ran successfully, so I'm changing this from draft to real PR. #15 depends on this for testing, will resume #15 when this is merged.

Removed Poetry, so we can just use Pip to install Python env. This still uses the `pyproject.toml` file. You install with either `pip install .` or `pip install .[test]` (`test` because that's what I called the section in the `pyproject.toml` file; we can name it something else like `dev`). This also entailed switching to `setuptools`, but the upshot is that we can use [setuptools_scm](https://github.com/pypa/setuptools_scm) to set the version number dynamically via e.g. Github tags.

I have updated `package.json` with `--host` in `npm run dev`, to ensure Vite forwards to the local machine.

I have updated the readme, so a new dev can hopefully get up and running.

I have changed the Github Actions workflow to make use of this dev container setup to run the backend CI.

The new files are:
* `./docker/Dockerfile.backend.dev`: contains the build instructions for the backend container. It has a default `CMD` which brings up the FastAPI backend server. This is overriden in the compose file
* `./docker/Dockerfile.frontend.dev`: contains the build instructions for the frontend container. It has a default `CMD` which runs `npm run dev`. This is overriden in the compose file
* `docker-compose.dev.yml`: the dev compose file.

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -5,96 +5,22 @@ on:
...
-          path: ./backend/.venv
-          key: pydeps-${{ hashFiles('**/poetry.lock') }}
-      - name: Install dependencies
-        run: poetry install --no-interaction --no-root
-        if: steps.cache-deps.outputs.cache-hit != 'true'
-      - name: Install project
-        run: poetry install --no-interaction
-      - name: Run tests
-        run: poetry run pytest
+      - name: Build and run containers
```
Comment (Duckling92) [inline .github/workflows/tests.yml:14]: Does this auto-build or is there some caching going on? If not, add `--build` imo.

File: backend/pyproject.toml [hunk 2]
```diff
@@ -48,6 +49,7 @@ line-length = 88
 target-version = "py311"
 select = ["ALL"]
 ignore = ["D100", "D104"]
+include = ["./backend/tests", "./backend/api"]
```
Comment (Duckling92) [inline backend/pyproject.toml:52]: pyproject.toml is in the `backend`-folder, so why is this not `include = ["./tests", "./api"]`?

File: .github/workflows/tests.yml (hunk 1 above)
Comment (ehrhorn) [reply .github/workflows/tests.yml:14]: Agreed, thank you, done.

File: backend/pyproject.toml (hunk 2 above)
Comment (ehrhorn) [reply backend/pyproject.toml:52]: Agreed, thank you, done.

**Review by Duckling92 — APPROVED** [review]

**Pull request merged**